Argument 1 to "apply" of "Series" has incompatible type "Callable[[Any], Union[int, NAType]]"

**To Reproduce**
1. Provide a minimal runnable `pandas` example that is not properly checked by the stubs.
```python
import pandas as pd

# these values are allowed
allowed = ["A", "B", "C"]

df = pd.DataFrame()

# we add a few that are not allowed
df["Column"] = ["A", "B", "C", "D"]

# replace values by index, and NA if not allowed
df["Label"] = df.Column.apply(lambda x: allowed.index(x) if x in allowed else pd.NA)

print(df.Label)
```

2. Indicate which type checker you are using (`mypy` or  `pyright`).
`mypy 1.2.0 (compiled: yes)`

3. Show the error message received from that type checker while checking your example.
```
bug.py:12: error: Argument 1 to "apply" of "Series" has incompatible type "Callable[[Any], Union[int, NAType]]"; expected "Callable[..., Union[str, bytes, date, datetime, timedelta, datetime64, timedelta64, bool, int, float, Timestamp, Timedelta, integer[Any], floating[Any], complex, Sequence[Any], Set[Any], Mapping[Any, Any]]]"  [arg-type]
bug.py:12: error: Incompatible return value type (got "Union[int, NAType]", expected "Union[str, bytes, date, datetime, timedelta, datetime64, timedelta64, bool, int, float, Timestamp, Timedelta, integer[Any], floating[Any], complex, Sequence[Any], Set[Any], Mapping[Any, Any]]")  [return-value]
Found 2 errors in 1 file (checked 1 source file)
```

**Please complete the following information:**
 - OS: Windows 11
 - OS Version: 22H2
 - python version: 3.10.11
 - version of type checker: 1.2.0
 - version of installed `pandas-stubs`: 2.0.1.230501

Related: #262?

--- a/pandas-stubs/core/series.pyi
+++ b/pandas-stubs/core/series.pyi
@@ -851,7 +851,7 @@ class Series(IndexOpsMixin, NDFrame, Generic[S1]):
     @overload
     def apply(
         self,
-        func: Callable[..., Scalar | Sequence | set | Mapping | None],
+        func: Callable[..., Scalar | Sequence | set | Mapping | NAType | None],
         convertDType: _bool = ...,
         args: tuple = ...,
         **kwds,

--- a/tests/test_series.py
+++ b/tests/test_series.py
@@ -35,6 +35,7 @@ from typing_extensions import (
 )
 import xarray as xr
 
+from pandas._libs.missing import NAType
 from pandas._libs.tslibs.timedeltas import Timedelta
 from pandas._libs.tslibs.timestamps import Timestamp
 from pandas._typing import (
@@ -469,6 +470,8 @@ def test_types_apply() -> None:
 
     ss = s.astype(str)
     check(assert_type(ss.apply(get_depth), pd.Series), pd.Series, np.int64)
+
+    check(assert_type(s.apply(lambda x: pd.NA), pd.Series), pd.Series, NAType)
 
 
 def test_types_element_wise_arithmetic() -> None:

